src/Symfony/Bridge/Doctrine/Tests/Form/Type/EntityTypeTest.php

```diff
@@ -1516,4 +1516,37 @@ public function testSubmitNullExpandedMultiple()
         $this->assertEquals($collection, $form->getNormData());
         $this->assertSame(array(), $form->getViewData(), 'View data is always an array');
     }
+
+    public function testSetDataEmptyArraySubmitNullMultiple()
+    {
+        $emptyArray = array();
+
+        $form = $this->factory->create(static::TESTED_TYPE, null, array(
+            'em' => 'default',
+            'class' => self::SINGLE_IDENT_CLASS,
+            'multiple' => true,
+        ));
+        $form->setData($emptyArray);
+        $form->submit(null);
+
+        $this->assertInternalType('array', $form->getData());
```

Comment:
@yceruto I think there is also a documentation issue here because it's said that returned value is an `array` http://symfony.com/doc/current/reference/forms/types/entity.html#multiple
Btw I'll change tests tomorrow